pkg/network/encoding/marshal/usm_http_test.go

```diff
@@ -307,6 +310,201 @@ func TestLocalhostScenario(t *testing.T) {
 	assert.Equal(uint32(1), aggregations.EndpointAggregations[0].StatsByStatusCode[int32(103)].Count)
 }
 
+// TestKubernetesLocalNATScenario tests how USMConnectionIndex handles Kubernetes-style local NAT connections
+// with both pre-NAT and post-NAT connections as seen in Kubernetes environments
+func TestKubernetesLocalNATScenario(t *testing.T) {
+	if runtime.GOOS == "windows" || os.Getenv("CI") == "true" {
+		t.Skip("Skipping test on Windows or CI")
+	}
```

Comment:
Consider moving the test to a new file with `linux` go build on it (as this test might run on macos)
also, why do we skip CI?